plugin/trino-mysql/src/test/java/io/trino/plugin/mysql/TestingMySqlServer.java

```diff
@@ -37,19 +40,14 @@ public TestingMySqlServer(boolean globalTransactionEnable)
 
     public TestingMySqlServer(String dockerImageName, boolean globalTransactionEnable)
     {
-        super(dockerImageName);
-        withDatabaseName("tpch");
+        MySQLContainer<?> container = new MySQLContainer<>(dockerImageName);
+        container = container.withDatabaseName("tpch");
         if (globalTransactionEnable) {
-            withCommand("--gtid-mode=ON", "--enforce-gtid-consistency=ON");
+            container = container.withCommand("--gtid-mode=ON", "--enforce-gtid-consistency=ON");
         }
-        start();
-        execute(format("GRANT ALL PRIVILEGES ON *.* TO '%s'", getUsername()), "root", getPassword());
-    }
-
-    @Override
-    public String getJdbcUrl()
-    {
-        return format("jdbc:mysql://%s:%s?useSSL=false&allowPublicKeyRetrieval=true", getContainerIpAddress(), getMappedPort(MYSQL_PORT));
+        this.container = container;
```

Comment:
Despite `this.container` is a reference, could you please move assignment to `this` to the very bottom of the constructor? That way it will be clear that everything above is preparation for the field.

Reply:
this is the last place where this object is modified. Lines below do not prepare this object. What is worse, they could even `container` to be up and ready.